src/coreclr/interpreter/compiler.cpp

```diff
@@ -4393,7 +4394,11 @@ void InterpCompiler::EmitCall(CORINFO_RESOLVED_TOKEN* pConstrainedToken, bool re
             {
                 // Tail call target needs to be IL
                 if (tailcall && isPInvoke && !isMarshaledPInvoke)
+                {
                     tailcall = false;
+                    // We need to inject ret after the call when we can't tail call
```

Comment:
Only inject the ret if we have a jmp instruction. tail.call should already have a ret in the right place, and tail.call allows more flexibility in the return value. (jmp requires that the types match... but a tail.call can in theory be a bit more flexible. (I don't think it is right now, but ... I don't think we need to experiment.

Reply:
Done